Community Toolbox/scripts/tests_ArrayMaxTests/tests_ArrayMaxTests.gml

```diff
@@ -19,19 +24,60 @@ function ArrayMaxTests(_run, _method) : VerrificMethodTest(_run, _method) constr
         then_result().should_be(10);
     }
     
+    static should_handle_large_array = function() {
+        given_array(array_with_100k_items);
+        when_array_max_runs();
+        then_result().should_be(99_999);
+    }
+    
+    static should_handle_positive_offset_and_length = function() {
+        given_array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+        given_offset_and_length(3, 4); // should take [4, 5, 6, 7]
+        when_array_max_runs();
+        then_result().should_be(7);
+    }
+    
+    static should_handle_negative_offset = function() {
+        given_array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+        given_offset_and_length(-5, 3); // should take [6, 7, 8]
+        when_array_max_runs();
+        then_result().should_be(8);
+    }
+    
+    static should_handle_negative_length = function() {
+        given_array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+        given_offset_and_length(5, -3); // should take [6, 5, 4]
+        when_array_max_runs();
+        then_result().should_be(6);
+    }
+    
+    static should_handle_large_subsection = function() {
+        given_array(array_with_100k_items);
+        given_offset_and_length(20_000, 70_000); // should take [20000, 20001, ..., 89998, 89999]
+        when_array_max_runs();
+        then_result().should_be(89_999);
+    }
+    
     // -----
     // Setup
     // -----
     
     array = [];
+    offset = 0;
```

Comment:
Is this fair? Passing a zero for the offset is different than letting the default offset take effect. (If the default were to change in the future, for instance to be `undefined` rather than 0, then this test could rapidly become stale). I might write `when_array_max_runs` to look something like

```
static when_array_max_runs = function() {
    if (is_undefined(offset)) {
        result = array_max(array);
    } else if (is_undefined(length)) {
        result = array_max(array, offset);
    } else {
        result = array_max(array, offset, length);
    }
}
```

then, of course, write tests calling it with one, two, and three arguments.

Reply:
Updated.